don't draw mnemonics in controls in wxUniv

Wlodzimierz-ABX-Skiba opened the pull request:
I don't know when and where the regression happened but in the past mnemonics weren't drawn in wxUniv builds. This patch fixes it.

Before patch it looks as follow: 
![wxUniv_broken_mnemonics](https://github.com/user-attachments/assets/f57042ec-c1fc-4aab-bc25-1ffafb473dce)

After patch if looks as follow:
![wxUniv_fixed_mnemonics](https://github.com/user-attachments/assets/8cc48e43-4bf9-485e-bcf0-8004ed30dcd2)

Wlodzimierz-ABX-Skiba committed:
don't draw mnemonics in contols in wxUniv

vadz commented:
I'm not sure about this one, it seems wrong to pass the label containing mnemonics to a function taking `indexAccel` in the first place. I.e. we should work with _either_ strings with "&" _or_ with pairs of string-without-"&" and accel index, but here we're mixing them up.

IOW I think we need to modify `wxControlRenderer::DrawButtonLabel()` and similar functions instead to pass `GetLabelText()` to the renderer functions, what do you think?

BTW, we could also override `GetLabelText()` in wxUniv to just return `m_label` as a micro-optimization.

Wlodzimierz-ABX-Skiba committed:
Merge branch 'wxWidgets:master' into wxUniv-fixed-drawing-mnemonics-in-controls

Wlodzimierz-ABX-Skiba committed:
that's wrong place for fixing it, revert changes

Wlodzimierz-ABX-Skiba committed:
microptimisation solves problem

wxControlBase::SetLabelText does EscapeMnemonics() so internally all & are && and later in drawing && are restored to & and that's why it was wrongly drawn. Microoptimisation in GetLabelText didn't work, only GetLabel does the trick

Wlodzimierz-ABX-Skiba commented:
wxControlBase::SetLabelText does EscapeMnemonics() so internally all & are && and later in drawing && are restored to & and that's why it was wrongly drawn. Microoptimisation in GetLabelText didn't work, only GetLabel does the trick

Wlodzimierz-ABX-Skiba commented:
Any idea why wxX11 build failed on all headers test? I did change header but wxMSWUniv build went fine

vadz commented:
Looks like a problem on GitHub Actions, the PCH file is not supposed to disappear like this :-( I've relaunched the job.

vadz commented on include/wx/univ/control.h:
```diff
@@ -65,7 +65,7 @@ class WXDLLIMPEXP_CORE wxControl : public wxControlBase, public wxInputConsumer
 
     // return the current label
-    virtual wxString GetLabel() const override { return wxControlBase::GetLabel(); }
+    virtual wxString GetLabel() const override { return m_label; }
```
Sorry, but this is wrong. `GetLabel()` should return the label _with_ mnemonics (i.e. `&Foo`), but `m_label` shouldn't contain them, i.e. it should be returned by `GetLabelText()`.

This should probably break a number of unit tests too — except that we don't even run them for wxUniv, I think, because so many of them fail...

Wlodzimierz-ABX-Skiba committed:
bring back GetLabel with mnemonics and move microoptimisation to GetLabelText

Wlodzimierz-ABX-Skiba committed:
here is the place where wrongly GetLabel instead of GetLabelText was called

Wlodzimierz-ABX-Skiba committed:
here is the place where wrongly GetLabel instead of GetLabelText was called

Wlodzimierz-ABX-Skiba commented:
Thanks, the problem was in renderer which used GetLabel instead of GetLabelText

vadz commented:
Looks good now, thanks, will merge soon.
